extensions/arc/src/ui/tree/controllerTreeNode.ts

```diff
@@ -51,21 +53,44 @@ export class ControllerTreeNode extends TreeNode {
 	}
 
 	private refreshChildren(registrations: Registration[]): void {
-		this._children = <TreeNode[]>registrations.map(registration => {
+		const newChildren: ResourceTreeNode[] = [];
+		registrations.forEach(registration => {
 			if (!registration.instanceNamespace || !registration.instanceName) {
 				console.warn('Registration is missing required namespace and name values, skipping');
-				return undefined;
+				return;
 			}
-			switch (registration.instanceType) {
-				case ResourceType.postgresInstances:
-					const postgresModel = new PostgresModel(this.model.info.url, this.model.auth!, registration.instanceNamespace, parseInstanceName(registration.instanceName));
-					return new PostgresTreeNode(postgresModel, this.model, this._context);
-				case ResourceType.sqlManagedInstances:
-					const miaaModel = new MiaaModel(this.model.info.url, this.model.auth!, registration.instanceNamespace, parseInstanceName(registration.instanceName));
-					return new MiaaTreeNode(miaaModel, this.model);
+
+			const resourceInfo: ResourceInfo = {
+				namespace: registration.instanceNamespace,
+				name: parseInstanceName(registration.instanceName)
+			};
+
+			let node = this._children.find(n => n.model?.info?.name === resourceInfo.name && n.model?.info?.namespace === resourceInfo.namespace);
```

Comment:
Could this filter for the instance type too?  It's possible to have a pg and miaa with the same namespace and name.

Reply:
Actually I'm not sure you can in practice.  But it'd be good to filter on.

Reply:
Yeah I think the namespace/name are at the k8 level so probably can't. Still easy enough to add so might as well.

Reply:
k8s lets you have resources with the same name and namespace if the resources are different types (which is the case for sql and pg).